synapse/tools/feed.py

```diff
@@ -47,38 +47,75 @@ def getItems(*paths):
             logger.warning('Unsupported file path: [%s]', path)
     return items
 
-async def addFeedData(core, outp, debug=False, *paths, chunksize=1000, offset=0, viewiden=None):
+async def ingest_items(core, items, outp, path, bname, viewiden=None, offset=None, chunksize=1000, is_synnode3=False, meta=None, debug=False):
+    tick = time.time()
+    outp.printf(f'Adding items from [{path}]')
+    foff = -1
+
+    for chunk in s_common.chunks(items, chunksize):
+        clen = len(chunk)
+        if offset and foff + clen <= offset:
+            foff += clen
+            continue
+        if is_synnode3 and meta is not None:
+            chunk = (meta,) + chunk
+        await core.addFeedData(chunk, viewiden=viewiden, reqmeta=is_synnode3)
+        foff += clen
+        outp.printf(f'Added [{clen}] items from [{bname}] - offset [{foff}]')
+    tock = time.time()
+    outp.printf(f'Done consuming from [{bname}]')
+    outp.printf(f'Took [{tock - tick}] seconds.')
+    if debug: # pragma: no cover
+        await s_t_storm.runItemStorm(core, outp=outp)
```

Comment:
Can we move this `if debug: ...` statement to the bottom of `addFeedData()` ? If someone specified multiple we'd drop into the Storm runtime after each item was parsed; OR we'd NEVER enter the runtime if someone didn't provide any items ( but still wanted to drop into a throwaway cortex ).